File: examples/php-openai/test_transport.php
<?php
declare(strict_types=1);
require __DIR__ . '/Transport.php';

$library = $argv[1] ?? throw new RuntimeException('Pass the staging native library path');
$transport = new Transport($library, ['environment' => 'staging', 'security' => 'e2ee']);
if (parse_url($transport->baseUrl, PHP_URL_HOST) !== '127.0.0.1') {
    throw new RuntimeException('Expected a local verifier proxy');
}
try {
    serialize($transport);
    throw new RuntimeException('Native handle was serialized');
} catch (LogicException) {
}
$transport->close();
$transport->close();
unset($transport);
try {
    new Transport($library, ['security' => 'unknown']);
    throw new LogicException('Invalid configuration was accepted');
} catch (RuntimeException) {
}
// A failed start must not prevent another start or retain the previous handle.
$next = new Transport($library, ['environment' => 'staging']);
$next->close();
echo "native ownership and configuration passed\n";

File: examples/luajit-http/test_transport.lua
package.path = (arg[0]:match("^(.*[/\\])") or "./") .. "?.lua;" .. package.path
local start = require "transport"
local library = assert(arg[1], "Pass the staging native library path")
local transport = start(library, { environment = "staging", security = "e2ee" })
assert(transport.base_url:match("^http://127%.0%.0%.1:"))
transport.close()
transport.close()
assert(not pcall(start, library, { security = "unknown" }))
start(library, { environment = "staging" }).close()
collectgarbage()
print("native ownership and configuration passed")
